app/models/steps/prefill.rb

```diff
@@ -0,0 +1,37 @@
+module Steps
+  class Prefill < Steps::Base
+    include CommodityHelper
+
+    def next_step_path
+      return duty_path if user_session.ni_to_gb_route? || user_session.eu_to_ni_route?
+
+      return next_step_for_gb_to_ni if user_session.gb_to_ni_route?
+
+      return next_step_for_row_to_gb if user_session.row_to_gb_route?
+
+      next_step_for_row_to_ni
+    end
+
+    private
+
+    def next_step_for_gb_to_ni
+      return interstitial_path if user_session.trade_defence
+      return duty_path if user_session.zero_mfn_duty
+
+      trader_scheme_path
+    end
+
+    def next_step_for_row_to_ni
+      return interstitial_path if user_session.trade_defence
+      return trader_scheme_path unless user_session.zero_mfn_duty
+
+      user_session.commodity_source = 'uk'
```

Comment:
this could create complications, 
this method is only to return the next step right? 
it should not change the commodity source (at least not inside this method)

Reply:
This is an idiom we've followed elsewhere, sadly. Possibly we could take the side effect out of the routing but we'd have to ask the routing questions twice